### PR TITLE
Use the AzDO Agent's copy of git

### PR DESCRIPTION
**Pull request by JoeRobich**
Try to use the AzDO agents copy of git to perform the shallow checkout.

### CODE REVIEW

File: eng/pipelines/checkout-windows-task.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ steps:
 
   - script: |
       @echo on
+      set PATH=$(Agent.HomeDirectory)\externals\git\cmd;"%PATH%
```
Comment (RikkiGibson) [inline eng/pipelines/checkout-windows-task.yml:7]: It looks like there's an unmatched `"` here. I wasn't sure if that was deliberate.

**Comment by RikkiGibson** [comment]
Will be happy to drive this but might want to wait until main is reopened.

File: eng/pipelines/checkout-windows-task.yml (hunk 1 above)
Comment (JoeRobich) [reply eng/pipelines/checkout-windows-task.yml:7]: This is what I get for typing in the web editor.

**Comment by JoeRobich** [comment]
> Will be happy to drive this but might want to wait until main is reopened.

Let's flow it as high as it will go.

**Pull request merged**